### ISSUE TITLE
NavigationView.ContainerFromMenuItem() returns null except when called from an event handler

### ISSUE THREAD

**Issue by rcohn** [microsoft/microsoft-ui-xaml#7645]
Hello -

I am developing a WinUI3 app containing a NavigationView.  The NV uses an observable collection to populate the NavigationViewItems.

The app must support expanding and collapsing NavigationViewItems programmatically.  The only way I know of to do that is to set NavigationViewItem.IsExpanded = true.  And the only way I know of to get the NavigationViewItem is through NavigationView.ContainerFromMenuItem().

The trouble is, ContainerFromMenuItem always returns null except when it is executed from a button event handler (such as a Click handler).  The app must perform expand/collapse without user input.

For example, I have a button that launches a Click event with this code, which works just fine to toggle an NVItem:

			Category selectedItem = (Category)navview.SelectedItem;
			int idx1 = Categories.IndexOf(selectedItem);
			var container = (NavigationViewItem)NavView.ContainerFromMenuItem(Categories[idx1]);
			if (container != null)
			{
				container.IsExpanded = !container.IsExpanded;
			}

However, that same code, when executed during app startup, such as in the MainWindow constructor after some test items are created, or in Attach(), always results in container being null.

So what am I doing wrong?

Thanks very much for any assistance on this.

**Comment by bkudiess** [comment]
The containers get created asynchronously, so they do not exist there yet, when the constructor runs. 
You can create your own NavViewItems and add them to the observable collection. That way you control the creation of NavViewItems yourself.

**Comment by rcohn** [comment]
Hello -

Giving the constructor example was bad (and misleading); my apologies. I do, of course, create NavViewItems dynamically.   The app, at certain times is controlled by external events to select items from the NavView lists.  It is really at those times that I'm trying to select the item and expand it.  As was suggested in another venue, setting a binding to the NavigationViewItem IsExpanded property is the way to do it.  I have tried that, but that binding seems not to ever have any effect (yes, it is an INotifyProperty value).

I did find a hacky way to achieve my goal.  That is to expand the selected item in the NavView's UpdateLayout handler.  That works as ContainerFromMenuItem now returns a valid NavViewItem.  But, that's not a great solution.

Thanks very much for responding.
